scheduler/pkg/kafka/gateway/client.go

```diff
@@ -153,14 +153,19 @@ func (kc *KafkaSchedulerClient) SubscribeModelEvents() error {
 
 		logger.Infof("Received event name %s version %d state %s", event.ModelName, latestVersionStatus.Version, latestVersionStatus.State.State.String())
 
-		switch latestVersionStatus.State.State {
-		case scheduler.ModelStatus_ModelAvailable:
+		// if there are available replicas then we add the consumer for the model
+		// note that this will also get triggered if the model is already added and in this case it is a no-op
+		if latestVersionStatus.State.GetAvailableReplicas() > 0 {
```

Comment:
I think this is entirely reasonable. The only question in my mind is whether we should deal differently with the case where the `availableReplicas < minReplicas`. Under the principle that if we can serve traffic we should, the > 0 condition is fine. The risk here is that we're going to overload the available replicas with traffic, but I guess autoscaling should kick in at that point...

Reply:
While I can change it check for this condition that we are above min replicas, I think min replicas is specifically use for now for scaling down decisions and not necessarily representing requiring min load. I suggest to keep it as it is and will add a note.